src/Spice86/UserControls/AddressAutoCompleteBox.axaml.cs

```diff
@@ -0,0 +1,128 @@
+    }
+
+    private void OnTextChanged(object? sender, TextChangedEventArgs e) {
+        OnTextChanged(AddressTextBox.Text);
+    }
+
+    private bool TryParseMemoryAddress(string? addressExpression,
+        [NotNullWhen(true)] out uint? address) {
+        if (string.IsNullOrWhiteSpace(addressExpression)) {
+            address = null;
+            return false;
+        }
+
+        try {
+            if (addressExpression.Contains(':')) {
+                string[] split = addressExpression.Split(":");
+                string firstPart = split[0];
+                string secondPart = split[1];
+
+                if(State is not null) {
+                    firstPart = ReplaceRegisterNameWithValue(firstPart, State);
+                    secondPart = ReplaceRegisterNameWithValue(secondPart, State);
+                }
+
+                if (split.Length > 1 &&
+                    ushort.TryParse(firstPart, NumberStyles.HexNumber,
+                    CultureInfo.InvariantCulture, out ushort hexSegment) &&
+                    ushort.TryParse(secondPart, NumberStyles.HexNumber,
+                    CultureInfo.InvariantCulture, out ushort hexOffset)) {
+                    address = Math.Min(A20Gate.EndOfHighMemoryArea, MemoryUtils.
+                        ToPhysicalAddress(hexSegment, hexOffset));
+                    return true;
+                }
+
+                if (split.Length > 1 &&
+                    ushort.TryParse(firstPart, NumberStyles.Integer,
+                    CultureInfo.InvariantCulture, out ushort segment) &&
+                    ushort.TryParse(secondPart, NumberStyles.Integer,
+                    CultureInfo.InvariantCulture, out ushort offset)) {
+                    address = Math.Min(A20Gate.EndOfHighMemoryArea, MemoryUtils.
+                        ToPhysicalAddress(segment, offset));
+                    return true;
+                }
+            } else if (ulong.TryParse(addressExpression,
+                      CultureInfo.InvariantCulture, out ulong value)) {
+                address = (uint)value;
+                return true;
+            } else if (ulong.TryParse(addressExpression.AsSpan(
+                  addressExpression.StartsWith("0x") ? 2 : 0),
+                  NumberStyles.HexNumber, CultureInfo.InvariantCulture,
+                      out ulong hexValue)) {
+                address = (uint)hexValue;
+                return true;
+            }
+        } catch { }
+        address = null;
+        return false;
+    }
+
+    private string ReplaceRegisterNameWithValue(string part, State state) {
+        return part.ToUpperInvariant() switch {
+            "CS" => state.CS.ToString("X4"),
+            "IP" => state.IP.ToString("X4"),
+            "DS" => state.DS.ToString("X4"),
+            "ES" => state.ES.ToString("X4"),
+            "FS" => state.FS.ToString("X4"),
+            "GS" => state.GS.ToString("X4"),
+            "SS" => state.SS.ToString("X4"),
+            _ => part
```

Comment:
No love for the other registers?
`ES:DI`, `DS:SI` and `AX:DX` could be fairly common usecases I think.

Reply:
File removed.